SimG4Core/CustomPhysics/plugins/RHStopTracer.cc

```diff
@@ -31,32 +34,37 @@ RHStopTracer::RHStopTracer(edm::ParameterSet const & p) {
   produces< std::vector<float> >("StoppedParticlesMass");
   produces< std::vector<float> >("StoppedParticlesCharge");
 
-  mParticle = G4ParticleTable::GetParticleTable()->FindParticle(mTraceParticleName);
+  mTraceEnergy *= CLHEP::GeV;
+  rePartName = mTraceParticleName;
 
-  edm::LogInfo("SimG4CoreCustomPhysics") 
+  edm::LogInfo("SimG4CoreCustomPhysics")
     << "RHStopTracer::RHStopTracer " << mTraceParticleName 
     << " Eth(GeV)= " << mTraceEnergy;
-  mTraceEnergy *= CLHEP::GeV;
+
 }
 
 RHStopTracer::~RHStopTracer() {
 }
 
 void RHStopTracer::update (const BeginOfRun * fRun) {
-  LogDebug("SimG4CoreCustomPhysics") 
+  LogDebug("SimG4CoreCustomPhysics")
     << "RHStopTracer::update-> begin of the run " << (*fRun)()->GetRunID(); 
 }
 
 void RHStopTracer::update (const BeginOfEvent * fEvent) {
-  LogDebug("SimG4CoreCustomPhysics") 
+  LogDebug("SimG4CoreCustomPhysics")
     << "RHStopTracer::update-> begin of the event " << (*fEvent)()->GetEventID(); 
 }
 
 void RHStopTracer::update (const BeginOfTrack * fTrack) {
   const G4Track* track = (*fTrack)();
   const G4ParticleDefinition* part = track->GetDefinition();
-  if(part && part == mParticle &&  track->GetKineticEnergy() > mTraceEnergy) {
-    LogDebug("SimG4CoreCustomPhysics") 
+  std::string stringPartName = part->GetParticleName();
+  bool matched = false;
+  if( (abs(part->GetPDGEncoding())>minPdgId && abs(part->GetPDGEncoding())<maxPdgId) || abs(part->GetPDGEncoding())==otherPdgId )
```

Comment:
please use std::abs [and maybe std::abs(part->GetPDGEncoding()) should be cached to avoid the repeated calculation